Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -2917,10 +2918,39 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     bool m_isListeningForUserFacingStateChangeNotification { false };
 #endif
     bool m_allowsMediaDocumentInlinePlayback { false };
-    std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isAudibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isCapturingActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_openingAppLinkActivity;
+
+    struct ProcessActivityState {
+        WTF_MAKE_STRUCT_FAST_ALLOCATED;
+
+        WebPageProxy& m_page;
```

Comment:
Those data members are never used outside right? I think it would be nice to switch to a class and have the data members private.